qiskit/pulse/builder.py

```diff
@@ -769,39 +784,46 @@ def call_subroutine(
             self._compile_lazy_circuit()
             subroutine = self._compile_circuit(subroutine)
 
-        empty_subroutine = True
-        if isinstance(subroutine, Schedule):
-            if len(subroutine.instructions) > 0:
-                empty_subroutine = False
-        elif isinstance(subroutine, ScheduleBlock):
-            if len(subroutine.blocks) > 0:
-                empty_subroutine = False
-        else:
+        if not isinstance(subroutine, (Schedule, ScheduleBlock)):
             raise exceptions.PulseError(
                 f"Subroutine type {subroutine.__class__.__name__} is "
                 "not valid data format. Call QuantumCircuit, "
                 "Schedule, or ScheduleBlock."
             )
 
-        if not empty_subroutine:
-            param_value_map = {}
-            for param_name, assigned_value in kw_params.items():
-                param_objs = subroutine.get_parameters(param_name)
-                if len(param_objs) > 0:
-                    for param_obj in param_objs:
-                        param_value_map[param_obj] = assigned_value
-                else:
-                    raise exceptions.PulseError(
-                        f"Parameter {param_name} is not defined in the target subroutine. "
-                        f'{", ".join(map(str, subroutine.parameters))} can be specified.'
-                    )
-
-            if value_dict:
-                param_value_map.update(value_dict)
+        if len(subroutine) == 0:
+            return
 
-            call_def = instructions.Call(subroutine, param_value_map, name)
-
-            self.append_instruction(call_def)
+        # Create local parameter assignment
+        local_assignment = dict()
+        for param_name, value in kw_params.items():
+            params = subroutine.get_parameters(param_name)
+            if not params:
+                raise exceptions.PulseError(
+                    f"Parameter {param_name} is not defined in the target subroutine. "
+                    f'{", ".join(map(str, subroutine.parameters))} can be specified.'
+                )
+            for param in params:
+                local_assignment[param] = value
+        if value_dict:
+            local_assignment.update(value_dict)
```

Comment:
Yes, there could be collision but this happens only intentionally. I added warning just in case. https://github.com/Qiskit/qiskit-terra/pull/8005/commits/dcf59a99647f05b902edd8851ccf371e162592bf